src/dmd/traits.d

```diff
@@ -1981,9 +1981,25 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
     }
     if (e.ident == Id.getLocation)
     {
-        if (dim != 1)
-            return dimError(1);
+        if (dim > 2) {
+            e.error("`getLocation` accepts at most 2 parameters, not %zu", dim);
+            return ErrorExp.get();
+        }
+
         auto arg0 = (*e.args)[0];
+        bool useAbsolutePath = false;
+        auto useAbsolutePathArg = dim == 2 ? (*e.args)[1] : null;
+        if (auto expr = isExpression(useAbsolutePathArg))
```

Comment:
This `if` should also have an `else` branch that outputs an error in case `useAbsolutePathArg` is not an expression.